media-api/app/model/Collection.scala

```diff
@@ -0,0 +1,20 @@
+package model
+
+import play.api.libs.json._
+
+case class Collection(name: String, subCollections: List[Collection])
+object Collection {
+  def buildTree(rootName: String, input: List[List[String]]): Collection = {
+    def loop(input: List[List[String]]): List[Collection] = {
+      input
+        .filter(_.nonEmpty).groupBy(_.head)
+        .map { case (parentCollectionName, subCollections) =>
+          Collection(parentCollectionName, loop(subCollections.map(_.tail)))
+        }
+        .toList
+    }
+    Collection(rootName, loop(input))
+  }
+
+  implicit def nodeFormat: Format[Collection] = Json.format[Collection]
+}
```

Comment:
There's a slightly more academic approach [here](https://gist.github.com/jamesgorrie/760c66f256db6bebec02) - but I couldn't get it working with the Json serialisers.